pkg/rancher/rancher.go

```diff
@@ -289,6 +289,9 @@ func (r *Rancher) Start(ctx context.Context) error {
 		if err := dashboarddata.Add(ctx, r.Wrangler, localClusterEnabled(r.opts), r.opts.AddLocal == "false", r.opts.Embedded); err != nil {
 			return err
 		}
+		if err := runPreflightMigrations(r.Wrangler); err != nil {
```

Comment:
I'm not sure that this is running at the correct place to make this check, for a few reasons:
- Since this only runs on leader, the non-leader pods will continue to run successfully (though the goal of the PR seems to be that no controllers run in this case).
  -  This further makes it difficult for users to actually know what is happening, since only a single replica of Rancher will be crashing. 
- This runs after a wide variety of controllers have already started (including the auth controllers). This makes it somewhat ineffective in ensuring that older controllers don't run on active data.

Reply:
The intent here is the following:
- Detect if Rancher has been downgraded without the use of the Backup Restore Operator
- If it has, prevent leader controllers from running

The purpose of this "fix" is not so much to present an all encompassing solution, but mainly a way to indicate to users "Hey, there is a problem, and it requires your manual intervention. Are you sure you want to go through with this?". We wouldn't need to do this if we have versioned CRDs and conversion webhooks. Granted I'm biased, since the provisioning controllers are what will affect me, but I am open to any suggestions you may have on better approaches. I needed it there because I needed it before the `dashboard` controllers are registered, but I need the wrangler clients to exist.